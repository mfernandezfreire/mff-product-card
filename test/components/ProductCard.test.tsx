import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';

import { product1 } from '../data/producto';

const { act } = renderer;



describe('ProductImage', () => {
    test('Debe mostar el componente con la imagen personalizada', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}  >
                {
                    ({ count }) => (
                        <>
                            <h1>Product Card</h1>
                            <h1>{count}</h1>
                        </>
                    )
                }
            </ ProductCard >
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    test('debe incrementar el contador', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}  >
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card Testing</h1>
                            <span>{count}</span>
                            <button onClick={() => increaseBy(1)}></button>
                        </>
                    )
                }
            </ ProductCard >

        )

        let tree = wrapper.toJSON();

        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();
        })

        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1');

    })
})
