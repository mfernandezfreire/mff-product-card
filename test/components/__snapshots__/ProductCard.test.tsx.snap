// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostar el componente con la imagen personalizada 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <h1>
    0
  </h1>
</div>
`;

exports[`ProductImage debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card Testing
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
